CougSat1-PMIC/src/events/EventPeriodic.cpp

```diff
@@ -17,9 +17,629 @@
+	long BattCurrA = 0; // Current leaving Battery A
+	long BattCurrB = 0; // Current Leaving Battery B
+	long PLCurr = 0; // Temp variable that will hold a current level value
+	long temp1 = 0; // Temperature of Battery A
+	long temp2 = 0; // Temperature of Battery B
+	int Pnum = 0;
+	int i = 13;
+
+	int index = 0;
+	uint8_t result;
+
+	/**************************** Monitor voltages, current, and Temperature of the batteries*********/
+	
+	
+	/*******************************************
+	 * Monitor voltage levels of batteries.    *
+	 * Use nodeVBattA and nodeVBattB to get    *
+	 * access to getVoltage() to get voltage   *
+	 * readings. Balance voltage levels of the *
+	 * batteries so that each battery has the  *
+	 * same voltage level. 
+	 ****************************************/
+	result = nodeVBattA.getVoltage(&BattVoltA);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get voltage of Battery A");
+		return result;
+	}
+	
+	result = nodeVBattB.getVoltage(&BattVoltB);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get voltage of Battery B");
+		return result;
+	}
+
+	if (BattVoltA > BattVoltB)
+	{
+		// if voltage of battery A is greater than the voltage of battery of B, redirect current to voltage B
+		// interate through nodesPVOut[] and set each one to go to battery B
+
+		while (nodesPVOut[index] < 8)
+		{
+			nodesPVOut[index].setSwitch(false, true); // turn pathA off and pathB on
+			index++;
+		}
+	}
+	else if (BattVoltB > BattVoltA)
+	{
+		// else if voltage of battery B is greater than the voltage of battery of A, redirect current to voltage A
+		while (nodesPVOut[index] < 8)
+		{
+			nodesPVOut[index].setSwitch(true, false); // turn pathA on and pathB off
+			index++;
+		}
+	}
+	else
+	{
+		// else both batteries are balanced and redirect both 
+		while (nodesPVOut[index] < 8)
+		{
+			nodesPVOut[index].setSwitch(true, true); // turn pathA on and pathB on
+			index++;
+		}
+	}
+
+	
+	/*******************************************
+	 * Monitor Current of the Batteries 	   *
+	 * Use NodeBattA and NodeBattB to access   *
+	 * getCurrent() to get battery current     *
+	 * readings. If current is above 4A,       *
+	 * reduce payloads until current drops     *
+	 * below 4A or equals 4A            	   *
+	 ****************************************/
+	double BattPR;
+	double RegPR;
+	int priority1 = 0;
+	int priority2 = 0;
+	float X1 = 0; X2 = 0;
+	int x_index = 0; // index for PR3V3
+	int battI = 0; // index for PRBatt[]
+	result = nodeBattA.getCurrent(&BattCurrA);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get current exiting Batttery A");
+		return result;
+	}
+
+	result = nodeBattB.getCurrent(&BattCurrB);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get current exiting Battery A");
+		return result;
+	}
+
+
+	// monitor current and reduce payload distribution
+	// by turing off payloads that draw current 
+	if (BattCurrA > 4 && BattCurrB > 4)
+	{
```

Comment:
Not sure what you are doing here and variable names and comments are not helping